atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/TransactionScopedCache.java

```diff
@@ -93,4 +104,18 @@ NavigableMap<byte[], RowResult<byte[]>> getRows(
     HitDigest getHitDigest();
 
     TransactionScopedCache createReadOnlyCache(CommitUpdate commitUpdate);
+
+    @Value.Immutable
+    interface CacheLookupResult {
+        Map<Cell, CacheValue> cacheHits();
+
+        Set<Cell> missedCells();
+
+        static CacheLookupResult of(Map<Cell, CacheValue> cachedValues, Set<Cell> missedCells) {
+            return ImmutableCacheLookupResult.builder()
+                    .cacheHits(cachedValues)
+                    .missedCells(missedCells)
+                    .build();
+        }
+    }
```

Comment:
This already existed, just moved from one of the implementations of the interface to here, so it's visible to everyone that needs it.